File: frontend/lib/modules/logistica/models/pedido.dart
import 'pedido_detalle.dart';

class Pedido {
  final int id;
  final int clienteId;
  final int? rutaId;
  final int vendedorId;
  final int? facturadorId;
  final String estado;
  final String? comentario;
  final double total;
  final DateTime createdAt;
  
  // Relations mapped from JSON
  final String? clienteNombre;
  final String? rutaNombre;
  final String? vendedorNombre;
  final List<PedidoDetalle> detalles;

  Pedido({
    required this.id,
    required this.clienteId,
    this.rutaId,
    required this.vendedorId,
    this.facturadorId,
    required this.estado,
    this.comentario,
    required this.total,
    required this.createdAt,
    this.clienteNombre,
    this.rutaNombre,
    this.vendedorNombre,
    this.detalles = const [],
  });

  factory Pedido.fromJson(Map<String, dynamic> json) {
    var list = json['detalles'] as List? ?? [];
    List<PedidoDetalle> detallesList = list.map((i) => PedidoDetalle.fromJson(i)).toList();

    return Pedido(
      id: json['id'],
      clienteId: json['cliente_id'],
      rutaId: json['ruta_id'],
      vendedorId: json['vendedor_id'],
      facturadorId: json['facturador_id'],
      estado: json['estado'],
      comentario: json['comentario'],
      total: double.tryParse(json['total']?.toString() ?? '0') ?? 0,
      createdAt: DateTime.parse(json['created_at']),
      clienteNombre: json['cliente']?['nombre'],
      rutaNombre: json['ruta']?['nombre'],
      vendedorNombre: json['vendedor']?['name'],
      detalles: detallesList,
    );
  }
}

File: backend/app/modules/Pedido/Models/Pedido.php
<?php

namespace App\Modules\Pedido\Models;

use App\Modules\Pedido\Enums\EstadoPedido;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Pedido extends Model
{
    use HasFactory;

    protected $fillable = [
        'cliente_id',
        'ruta_id',
        'vendedor_id',
        'facturador_id',
        'estado',
        'comentario',
        'total',
        'latitud',
        'longitud',
        'fecha_entrega',
        'nota_produccion',
    ];

    protected $casts = [
        'estado' => EstadoPedido::class,
        'total' => 'decimal:2',
        'latitud' => 'decimal:8',
        'longitud' => 'decimal:8',
        'fecha_entrega' => 'date',
    ];

    public function cliente()
    {
        return $this->belongsTo(LedhouseCliente::class, 'cliente_id');
    }

    public function ruta()
    {
        return $this->belongsTo(Ruta::class, 'ruta_id');
    }

    public function vendedor()
    {
        return $this->belongsTo(User::class, 'vendedor_id');
    }

    public function facturador()
    {
        return $this->belongsTo(User::class, 'facturador_id');
    }

    public function detalles()
    {
        return $this->hasMany(PedidoDetalle::class, 'pedido_id');
    }

    public function despachos()
    {
        return $this->belongsToMany(Despacho::class, 'despacho_pedido')
            ->withPivot(['orden_entrega', 'estado_entrega', 'comentario_entrega'])
            ->withTimestamps();
    }
}
